### PR TITLE
Pre-support for C++20

### PR DESCRIPTION
**Pull request by fuzhe1989**
### What problem does this PR solve?

Issue Number: ref #5993

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/Page/V3/PageEntriesEdit.h [hunk 1]
```diff
@@ -65,6 +65,32 @@ struct PageVersion
         return sequence <= rhs.sequence;
     }
 };
+} // namespace DB::PS::V3
+/// See https://fmt.dev/latest/api.html#formatting-user-defined-types
```
Comment (JaySon-Huang) [inline dbms/src/Storages/Page/V3/PageEntriesEdit.h:69]: why we need to move it up here?

Comment (fuzhe1989) [reply dbms/src/Storages/Page/V3/PageEntriesEdit.h:69]: "Instantiation after usage"

File: CMakeLists.txt [hunk 2]
```diff
@@ -276,9 +276,9 @@ endif ()
 #endif ()
 
 if (CMAKE_VERSION VERSION_LESS "3.8.0")
-    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++1z")
+    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++20")
```
Comment (fuzhe1989) [inline CMakeLists.txt:279]:
```suggestion
    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++1z")
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -276,9 +276,9 @@ endif ()
...
 else ()
-    set (CMAKE_CXX_STANDARD 17)
+    set (CMAKE_CXX_STANDARD 20)
```
Comment (fuzhe1989) [inline CMakeLists.txt:281]:
```suggestion
    set (CMAKE_CXX_STANDARD 17)
```

**Review by ywqzzy — APPROVED** [review]

File: dbms/src/Storages/Page/V3/PageEntriesEdit.h (hunk 1 above)
Comment (JaySon-Huang) [reply dbms/src/Storages/Page/V3/PageEntriesEdit.h:69]: I get it

**Pull request merged**